uuid-cluster-dns/README.md

```diff
@@ -0,0 +1,60 @@
+# Using UUIDs for cluster endpoint domains
+
+The current approach for building cluster domains is to include the workload cluster's ID as a subdomain of the management cluster it belongs to, e.g.:
+
+```
+https://api.j9yr2.k8s.gauss.eu-west-1.aws.gigantic.io
+```
+
+This approach has a few drawbacks:
+
+1. The workload cluster is tied to the management cluster, making migration messy
+2. Workload clusters cannot be "promoted" to management clusters as the domain would no longer make sense
+3. Several customers have their own domain set up, leading to inconsistency between MCs (e.g. `eu-central-1.aws.cps.customer.com` vs. `gauss.eu-west-1.aws.gigantic.io`)
+4. The provider (and region) is baked into the domain, effectively preventing multi-region (or multi-cloud)
+
+An alternative "flat" DNS structure has been proposed that removes the link between WC and MC. E.g.:
+
+```
+https://api.j9yr2.k8s.[customer name].gigantic.io
+```
+
+While this solves some of the above drawbacks (1, 2, 4) it introduces its own complications:
+
+1. The customer name (or codename) is exposed in the domain
+2. There is no enforced global uniqueness on cluster IDs so if a customer has multiple MCs there is a chance that two workload clusters have the same ID, leading to conflicting domains
+
+## Proposal
+
+> Assumption: this is relevant only going forward with CAPI clusters and not to be "backported" to the vintage product.
+
+The flat DNS structure can be improved to avoid the drawbacks above my leveraging a UUID for the cluster domain and dissasociate the cluster ID/Name from URLs.
+
+E.g.
+
+```
+https://api.4c6aba87-061c-458d-91b6-62ad07979728.k8s.gigantic.io
```

Comment:
My understanding of the flat DNS structure was that we would still have a hosted zone per customer. At least that was what @kopiczko proposed if I'm not mistaken. This would still allow us to convert a workload cluster into a management cluster for the same customer.

We in Phoenix talked about slightly changing this by having a hosted zone per customer and cloud provider because otherwise we would have to do cross cloud DNS configuration, which is rather complicated (e.g. credentials management).

Reply:
The problem with that though is you're (potentially) exposing the customer name and it's then an issue if the customer changes for any reason (I believe we've seen this with one of our customers haven't we?)

Reply:
but we dont need to put the real name there, just a unique name that serves to identify and discriminate between customers

Reply:
We were talking about "customer ID" which isn't really related to the name. We should also be able to assign more than one domain ID per customer.